direr.go

```diff
@@ -0,0 +1,42 @@
+package tests
+
+import (
+	"testing"
+
+	"github.com/google/uuid"
+	. "github.com/smartystreets/goconvey/convey"
+
+	"github.com/beyondstorage/go-storage/v4/types"
+)
+
+func TestDirer(t *testing.T, store types.Storager) {
+	Convey("Given a basic Storager", t, func() {
+		Convey("The Storager should implement Direr", func() {
+			_, ok := store.(types.Direr)
+			So(ok, ShouldBeTrue)
+		})
+
+		Convey("When CreateDir", func() {
```

Comment:
As we discussed in https://github.com/beyondstorage/go-storage/issues/560#issuecomment-853751578, do we need to reflect them in the tests?

Reply:
Should we add the following cases:
- `CreateDir` with `path` not end with `/`
- `CreateDir` with `path` end with `/`
- `CreateDir` with `workDir`
- `List` an empty dir
- `List` a dir within files
The results of creating an existing dir and file and directory with the same name are different between services, so we needn't to add the related cases?

Reply:
> CreateDir with path not end with /

LGTM

> CreateDir with path end with /

Service should return error, but the error should be defined.

> CreateDir with workDir

Maybe we should ignore `dir exists` related error here?

> List an empty dir
> List a dir within files

I think they are not related.

---

Here comes up a new problem, for object storage services: If we Stat(“abc”) after CreateDir("abc"), we will get a not exist. We need to find a way to figure it out.

Reply:
>If we Stat(“abc”) after CreateDir("abc"), we will get a not exist. We need to find a way to figure it out.
Return not exist when neither `abc` and `abc/` exist？
Same problem with `Delete`, we don't kown whether to delete the file or dir.

Reply:
>If we Stat(“abc”) after CreateDir("abc"), we will get a not exist. We need to find a way to figure it out.

Return not exist when neither abc and abc/ exist？
Same problem with Delete, we don't kown whether to delete the file or dir.

Reply:
>If we Stat(“abc”) after CreateDir("abc"), we will get a not exist. We need to find a way to figure it out.

Return not exist when neither `abc` nor `abc/` exist？
Same problem with `Delete`, we don't kown whether to delete the file or dir.

Reply:
Yes, it's indeed a problem.

Reply:
Could be resolved via https://github.com/beyondstorage/specs/pull/93